libs/gltfio/src/StbProvider.cpp

```diff
@@ -84,15 +90,35 @@ class StbProvider final : public TextureProvider {
     std::string mRecentPushMessage;
     std::string mRecentPopMessage;
     Engine* const mEngine;
+    const std::optional<unsigned int> mMaxTextureSize;
 };
 
+static int scaleDownDimension(int dimension, float scaleDownFactor) {
+    assert_invariant(dimension > 0);
+    const auto scaledDimension = std::max(1, static_cast<int>(std::round(static_cast<float>(dimension) / scaleDownFactor)));
+    return (scaledDimension);
+}
+
+static void scaleDownDimensionsIfNeeded(int &width, int &height, unsigned int maxTextureSize) {
```

Comment:
for this, I prefer something like:
```
std::pair<uint32_t, uint32_t> scaleDownDimensionsIfNeeded(uint32_t width, uint32_t height,, uint32_t maxSize);
```

rather than using in/out parameters.

Type should not be signed, and also we're using sized types in our APIs.

Reply:
ended up calling this `getFinalDimensions` with the signature you suggested